test/common/http/BUILD

```diff
@@ -415,10 +415,11 @@ envoy_cc_test(
     ],
 )
 
+# TODO(mattklein123): Use compatible with when we move to Bazel 4.0.
 envoy_cc_test(
     name = "conn_pool_grid_test",
-    srcs = ["conn_pool_grid_test.cc"],
-    deps = [
+    srcs = envoy_select_enable_http3(["conn_pool_grid_test.cc"]),
```

Comment:
I think this one can stay as-is.  The test should be intrumented to run the HTTP/3 checks when HTTP/3 is on, and just test the grid (eventually the ipv4/ipv6 parts) when http/3 is off.

Reply:
Without this it didn't compile, so I suspect that at runtime the old code would have failed due to missing factories if compiled out? Can I live this and you can decide how you want the code to work without H3 compiled in since it's new code?

Reply:
Do you know which compile it failed?  Both the quic includes and libraries were already selected.
but yeah, feel free to toss in a TODO for me